chefkoch/recipe.py

```diff
@@ -0,0 +1,771 @@
+        logger.debug("Filepath (dict): " + str(filepath))
+        logger.debug("Key (dict): " + str(key))
+        self.key = key
+        logger.debug("Key: " + str(self.key))
+        if filepath is None:
+            raise IOError("The filepath is None.")
+            return
+        if os.path.isfile(filepath):
+            self.file = filepath
+            logger.debug("Filepath: " + str(self.file))
+        else:
+            logger.warning("The file " + filepath + " does not exist.")
+            raise IOError("The file " + filepath + " does not exist.")
+            return
+
+    def tostring(self):
+        """
+        Returns a printable and formatted string that shows the
+        FileParamValue and its values.
+
+        :returns:
+            string - Holds structured information on the FileParamValue
+        """
+        content = "Value is following file: \n  "
+        content+= self.file + "\n  Key: " + str(self.key)
+        return content
+
+
+class Param:
+    """
+    A parameter with all values attached to it.
+    """
+    # todo: make Param class extend list so that Param returns Param.values
+    values = [] #: All possible values of the parameter
+    
+    def __init__(self, name, entry):
+        """
+        Creates a new paramter from the JSON data gotten from the flavour file.
+        
+        :param name:        name as provided as in flavour file
+        :type name:         string
+        :param entry:       if the falvour file was a dict, it was flavour['name']
+        :type entry:        dictionary
+        """
+        self.values = []
+        logger.debug("Creating a new parameter " + str(name))
+        self.name = name
+        if type(entry) is not list: # a single value so to say
+            logger.debug("It has a single value: " + str(entry))
+            self.appendEntry(entry)
+        else:
+            logger.debug("It has more than one value.")
+            for sub_entry in entry: # the entry in the json file
+                self.appendEntry(sub_entry)
+        logger.debug("Fin.")
+
+    def appendFileParam(self, entry):
+        """
+        Appends a file parameter given in the JSON data to the Param.values
+        list.
+        
+        :param entry:    dict with fields type, file and key
+        :type entry:     dictionary
+        :raises ValueError: If there is no 'file' field to the entry or if the
+            given file path does not exist
+        """
+        try:
+            newValue = FileParamValue(entry['file'], entry['key'])
+            self.values.append(newValue)
+            logger.debug("Appending " + str(newValue) + newValue.tostring())
+        except KeyError as err:
+            # KeyError because the key-field is missing, in this case ignore
+            # todo: different possible exceptions
+            logger.exception("Either the file or the key field of the " +
+                             "entry are missing.")
```

Comment:
check, if that actually is nessessary. I believe a `logger` may be configured such, that it catches exceptions to the logs without having to bloat the code like this